### ISSUE TITLE
Keyboard becomes Visible When accessing Drawer

### ISSUE THREAD

**Issue by ram231** [MisterJimson/flutter_keyboard_visibility#28]
Hi. Have you tested this kind of output?
It is the same implementation as `KeyboardDismissOnTap` widget.
I might submit an issue to the flutter repository, but it is somehow related to this too.

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      home: KeyboardDismissOnTap(),
    );
  }
}
/// Tap the TextField first, then tap anywhere to close it.
/// Then Open the drawer in the appbar. the keyboard layout will show
/// despite the method has been unscoped.
class KeyboardDismissOnTap extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return GestureDetector(
      onTap: () {
        final currentScope = FocusScope.of(context);
        if (!currentScope.hasPrimaryFocus) {
          currentScope.unfocus();
        }
      },
      child: Scaffold(
        appBar: AppBar(),
        drawer: Drawer(),
        body: Column(
          children: [
            Text("Tap the textfield, then tap anywhere to close it, then tap on the drawer. The keyboard visibility will open"),
            TextField(),
                    ],
                  ),
                );
              },
            )
          ],
        ),
      ),
    );
  }
}

```

**Comment by matthew-carroll** [comment]
Thanks for filing. I'll try to reproduce this locally when I get a chance and see what the fix should be.

**Comment by ram231** [comment]
I'll put a reference in the flutter repository here
https://github.com/flutter/flutter/issues/59927

**Comment by ram231** [comment]
There is a current fix regarding this issue, I'm going to close this again, feel free to open this issue @matthew-carroll 

Ref: https://github.com/flutter/flutter/issues/54277#issuecomment-640998757

**Comment by MisterJimson** [comment]
Fixed in 3.2.1
